doc/book/migration/rc-to-v1.md

```diff
@@ -0,0 +1,359 @@
+            // Optional:
+            //    'path'  => '/path/to/match',
+            //    'error' => true,
+            //],
+            [
+                'middleware' => [
+                    Helper\ServerUrlMiddleware::class,
+                    Helper\UrlHelperMiddleware::class,
+                ],
+            ],
+        ],
+
+        // An array of middleware to register after registration of the
+        // routing middleware
+        'post_routing' => [
+            //[
+            // Required:
+            //    'middleware' => 'Name of middleware service, or a callable',
+            // Optional:
+            //    'path'  => '/path/to/match',
+            //    'error' => true,
+            //],
+        ],
+    ],
+]
+```
+
+### Impact
+
+While the configuration from RC5 and earlier will continue to work, it will
+raise deprecation notices. As such, you will need to update your configuration
+to follow the guidelines created with RC6.
+
+RC6 and later change the configuration to remove the `pre_routing` and
+`post_routing` keys. However, individual items within the array retain the same
+format as middleware inside those keys, namely:
+
+```php
+[
+    // Required:
+    'middleware' => 'Name of middleware service, or a callable',
+    // Optional:
+    //    'path'  => '/path/to/match',
+    //    'error' => true,
+]
+```
+
+Additionally, the routing and dispatch middleware now become items in the array;
+they (or equivalent entries for your own implementations) must be present in
+your configuration if you want your routed middleware to dispatch!  This change
+gives you full control over the flow of the pipeline.
+
+To specify the routing middleware, use the constant
+`Zend\Expressive\Container\ApplicationFactory::ROUTING_MIDDLEWARE` in place of
+a middleware array; this has the value `EXPRESSIVE_ROUTING_MIDDLEWARE`, if you
+do not want to import the class. Similarly, for the dispatch middleware, use the
+constant `Zend\Expressive\Container\ApplicationFactory::DISPATCH_MIDDLEWARE`
+(value `EXPRESSIVE_DISPATCH_MIDDLEWARE`) to specify the dispatch middleware.
+
+As such, the default configuration now becomes:
+
+```php
+return [
+    'middleware_pipeline' => [
+        // An array of middleware to pipe to the application.
+        // Each item is of the following structure:
+        // [
+        //     // Required:
+        //     'middleware' => 'Name or array of names of middleware services and/or callables',
+        //     // Optional:
+        //     'path'  => '/path/to/match',
+        //     'error' => true,
+        // ],
+        [
+            'middleware' => [
+                Helper\ServerUrlMiddleware::class,
+                Helper\UrlHelperMiddleware::class,
+            ],
+        ],
+
+        // The following is an entry for the routing middleware:
+        Zend\Expressive\Container\ApplicationFactory::ROUTING_MIDDLEWARE,
+
```

Comment:
Could add an example here to illustrate that you can pipe a middleware between them?